pyston/utils/datastructures.py

```diff
@@ -34,14 +34,25 @@ def __ne__(self, other):
 
 class FieldsetGenerator:
 
-    def __init__(self, resource=None, fields_string=None):
+    def __init__(self, resource=None, fields_string=None, direct_serialization=False):
         self.resource = resource
         self.fields_string = fields_string
+        self.direct_serialization = direct_serialization
 
-    def _get_resource_class(self, obj):
-        from pyston.resource import typemapper
+    def _get_resource(self, obj):
+        from pyston.serializer import get_resource_or_none
 
-        return typemapper.get(obj)
+        if self.resource:
+            return get_resource_or_none(self.resource.request, obj, self.resource.resource_typemapper)
+        else:
+            return None
+
+    def _get_allowed_fieldset(self):
```

Comment:
added check of resource allowed fields. Therefore for only allowed fields are generated columns to the csv/xlsx